workspaces/ui-v2/Taskfile.yml

```diff
@@ -0,0 +1,85 @@
+version: "3"
+
+tasks:
+  clean:
+    desc: Removes UI's build directories
+    cmds:
+      - rm -rf build
+
+  ci:
+    - task: cloud_spec_viewer:ci
+
+  #
+  # docker namespace
+  #
+  docker:build:
+    desc: Package site as a Docker container
+    vars:
+      TAG:
+        sh: echo $(../../bin/clean-docker-tag {{.TAG}})
+    cmds:
+      - |
+        docker build . \
+          -t $DOCKER_REGISTRY/{{.REPO}}:{{.TAG}} \
+          --build-arg APP={{.REPO}}
+
+  docker:push:
+    vars:
+      TAG:
+        sh: echo $(../../bin/clean-docker-tag $TAG)
+    cmds:
+      - docker push $DOCKER_REGISTRY/{{.REPO}}:{{.TAG}}
+      - |
+        if [[ "{{.TAG}}" =~ ^release-b ]]
+        then
+          echo "tagging and pushing 'latest'"
+          docker tag $DOCKER_REGISTRY/{{.REPO}}:{{.TAG}} $DOCKER_REGISTRY/{{.REPO}}:latest
+          docker push $DOCKER_REGISTRY/{{.REPO}}:latest
+        fi
+
+  #
+  # cloud_spec_viewer namespace
+  #
+  cloud_spec_viewer:build:
+    desc: Build cloud_spec_viewer site
+    cmds:
+      - task: :workspaces:build
+      - >
+        BUILD_PATH=build-cloud_spec_viewer
+        REACT_APP_API_URL=https://api.useoptic.com
```

Comment:
I... _think_ this might be the place to start adding staging/prod builds? I really only very tenuously understand how all of this fits together.

Reply:
yep I believe the build process produces the docker artifact and publishes it to our container registry. we can then deploy that into our staging or production k8s per instructions I can send you in our slack

Reply:
post convo: optichub has an example of setting REACT_APP_* stuff based off branches as an environment proxy. here (or rather, where this was copied from), its being set in the taskfile because we only produced prod builds for UI _and_ there's no _deployment_ ci in monorail.

because i didnt want the optic repo to talk to our k8s cluster, the CI workflow in this repo builds and pushes to ECR.

in monorail, we define the helm chart (wrapped with taskfile), but there's no corresponding CI workflow there to do the deployment for you. the ui stuff is somewhat unique in the sense that it doesnt have automatic deployments set up. this part of the code base was slow moving enough it never became a thing we needed. can be changed for sure.